### ISSUE TITLE
[Tuning] AWS.CloudTrail.ResourceMadePublic false positive if bucket policy condition uses `aws:userId` without a wildcard

### ISSUE THREAD

**Issue by TonyLovesDevOps** [panther-labs/panther-analysis#1590]
### Rule ID

AWS.CloudTrail.ResourceMadePublic

### Description

See [https://github.com/Netflix-Skunkworks/policyuniverse/issues/189](https://github.com/Netflix-Skunkworks/policyuniverse/issues/189) for details, but this bucket policy triggers a false positive:

```json
{
  "Statement": [
    {
      "Action": [
        "s3:GetInventoryConfiguration",
        "s3:GetBucketLocation",
        "s3:GetBucketAcl"
      ],
      "Condition": {
        "StringLike": {
          "aws:userId": "AROAQXXXXXXYYYYYYZZZZ:foo"
        }
      },
      "Effect": "Allow",
      "Principal": {
        "AWS": "*"
      },
      "Resource": "arn:aws:s3:::mybucket"
    }
  ],
  "Version": "2012-10-17"
}
```

### Example Log (redacted)

```json
{
  "eventVersion": "1.11",
  "userIdentity": {
    "type": "AssumedRole",
    "principalId": "AROAXXXXXXXXEXAMPLE:john.doe@example.com",
    "arn": "arn:aws:sts::111122223333:assumed-role/ExampleRole/john.doe@example.com",
    "accountId": "111122223333",
    "accessKeyId": "ASIAxxxxxxxxEXAMPLE",
    "sessionContext": {
      "sessionIssuer": {
        "type": "Role",
        "principalId": "AROAXXXXXXXXEXAMPLE",
        "arn": "arn:aws:iam::111122223333:role/ExampleRole",
        "accountId": "111122223333",
        "userName": "ExampleRole"
      },
      "attributes": {
        "creationDate": "2025-04-15T20:16:55Z",
        "mfaAuthenticated": "false"
      }
    }
  },
  "eventTime": "2025-04-16T00:16:16Z",
  "eventSource": "s3.amazonaws.com",
  "eventName": "PutBucketPolicy",
  "awsRegion": "us-west-2",
  "sourceIPAddress": "203.0.113.12",
  "userAgent": "[Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/135.0.0.0 Safari/537.36]",
  "requestParameters": {
    "bucketPolicy": {
      "Version": "2012-10-17",
      "Statement": [
        {
          "Sid": "Foo",
          "Effect": "Allow",
          "Principal": {
            "AWS": "*"
          },
          "Action": [
            "s3:GetInventoryConfiguration",
            "s3:GetBucketLocation",
            "s3:GetBucketAcl"
          ],
          "Resource": "arn:aws:s3:::example-bucket",
          "Condition": {
            "StringLike": {
              "aws:userid": "AROAXXXXXXXXEXAMPLE:foo"
            }
          }
        }
      ]
    },
    "bucketName": "example-bucket",
    "Host": "example-bucket.s3.us-west-2.amazonaws.com",
    "policy": ""
  },
  "responseElements": null,
  "additionalEventData": {
    "SignatureVersion": "SigV4",
    "CipherSuite": "TLS_AES_128_GCM_SHA256",
    "bytesTransferredIn": 3321,
    "AuthenticationMethod": "AuthHeader",
    "x-amz-id-2": "REDACTED-BASE64-STRING",
    "bytesTransferredOut": 0
  },
  "requestID": "EXAMPLE123456789",
  "eventID": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
  "readOnly": false,
  "resources": [
    {
      "accountId": "111122223333",
      "type": "AWS::S3::Bucket",
      "ARN": "arn:aws:s3:::example-bucket"
    }
  ],
  "eventType": "AwsApiCall",
  "managementEvent": true,
  "recipientAccountId": "111122223333",
  "eventCategory": "Management",
  "tlsDetails": {
    "tlsVersion": "TLSv1.3",
    "cipherSuite": "TLS_AES_128_GCM_SHA256",
    "clientProvidedHostHeader": "example-bucket.s3.us-west-2.amazonaws.com"
  }
}
```

**Comment by arielkr256** [comment]
Hi @TonyLovesDevOps , can you confirm you're still seeing false positives after applying this update? https://github.com/panther-labs/panther-analysis/pull/1533

**Comment by TonyLovesDevOps** [comment]
@arielkr256 I'd like to do that. #1533 was merged on March 11 and I encountered this alert on April 15.  I haven't used panther much though, do we need to do anything to pull in rule updates?
